cudamapper/include/cudamapper/types.hpp

```diff
@@ -29,4 +31,35 @@ using representation_t = std::uint64_t; // this depends on kmer size, in some ca
 /// read_id_t
 using read_id_t = std::uint64_t; // can this be 32-bit?
 
+/// Anchor - represents one anchor
+///
+/// Anchor is a pair of two sketch elemetns with the same sketch element representation from different reads
+typedef struct Anchor{
+    // read id of first sketch element
+    read_id_t query_read_id_;
+    // read id of second sketch element
+    read_id_t target_read_id_;
+    // position of first sketch element in query_read_id_
+    position_in_read_t query_position_in_read_;
+    // position of second sketch element in target_read_id_
+    position_in_read_t target_position_in_read_;
+} Anchor;
+
+
+/// Overlap - represents one overlap between two substrings
```

Comment:
For each pair of reads there is only one overlap or this does not have to be the case?

Reply:
This does not technically need to be the case (though it does in our implementation), e.g if you have repeated DNA.

Reply:
Not sure I understand this, but marking as resolved. You can explain me the difference offline.